client/src/Components/HomePage/NewsPopUp/AddNewsPopUp.tsx

```diff
@@ -0,0 +1,128 @@
+import { Alert  } from 'react-native'
+import { observer } from 'mobx-react-lite';
+import { NavigationProp, useNavigation } from '@react-navigation/native';
+import { RootStackParamList } from '../../../types/navigation';
+import axios from "axios";
+import { NewsItem } from '../../../types/NewsItem';
+import CustomButton from '../../../CustomComponents/CustomButton';
+import CustomView from '../../../CustomComponents/CustomView';
+import CustomText from '../../../CustomComponents/CustomText';
+import CustomInput from '../../../CustomComponents/CustomInput';
+import getNewsStore from '../../../stores/newsStore';
+import getLoginStore from '../../../stores/loginStore';
+import getThemeStore from '../../../stores/themeStore';
+import { darkMode, lightMode } from '../../colors/colors';
+
+
+const AddNewsPopUp : React.FC = observer(() => {
+    const navigation = useNavigation<NavigationProp<RootStackParamList>>();
+    const handleSubmit = async () => {
+        if (getNewsStore().news.get()) {
+            try {
+                await handleAddNews();
+                const newsItem: NewsItem = {
+                    user_name: getLoginStore().user_name.get(),
+                    date_of_news: getNewsStore().date.get(),
+                    news_content: getNewsStore().news.get()
+                }
+                getNewsStore().addNews(newsItem);
+                Alert.alert('Success', 'News added successfully');
+                navigation.goBack();
+            } catch ( error ) {
+                console.log("Error in handling add news!" , error)
+            }      
+        } else {
+            Alert.alert('Error' , 'Please Enter the news');
+        }        
+    }
+
+    const handleAddNews = async () : Promise<void> => {
+        try {
+            const IP_ADDRESS : string = "192.168.100.126"
+            if( getLoginStore().user_id.get() !== -1)
+            {
+                await axios.post(`http://${IP_ADDRESS}:3000/api/addNews`, {
+                    user_id: getLoginStore().user_id.get(),
+                    news_content : getNewsStore().news.get()
+                });
+            }
+        } catch ( error ) {
+            console.error("Error while adding News" , error);
+        }
+    }
+    const theme = getThemeStore().isDarkThemeEnabled.get() ? darkMode : lightMode;
+
+   return (
+        <CustomView
```

Comment:
split this component into small components

Reply:
DONE ✅